x-pack/plugins/apm/server/plugin.ts

```diff
@@ -46,14 +46,17 @@ export class APMPlugin implements Plugin<APMPluginContract> {
       createApmApi().init(core, { config$: mergedConfig$, logger, __LEGACY });
     });
 
-    combineLatest(mergedConfig$, core.elasticsearch.dataClient$).subscribe(
-      ([config, dataClient]) => {
-        createApmAgentConfigurationIndex({
-          esClient: dataClient,
-          config,
-        });
-      }
-    );
+    await new Promise(resolve => {
+      combineLatest(mergedConfig$, core.elasticsearch.dataClient$).subscribe(
+        async ([config, dataClient]) => {
+          await createApmAgentConfigurationIndex({
+            esClient: dataClient,
+            config,
+          });
+          resolve();
+        }
+      );
+    });
```

Comment:
Without knowing the rxjs api well, can you use the `toPromise` method here, instead of the promise constructor?

Reply:
`createAgentConfiguration` is in `subscribe`, which deals with side-effects. It is not an observable itself, so we cannot use `toPromise()` unless we create another observable that we complete when the agent configuration has been created. Which is possible but I figured it would be overkill.